### PR TITLE
Teambuilder: Debounce searching teams

### PR DESCRIPTION
**Pull request by PartMan7**
Wasn't fun having my browser frozen for half a minute every time I typed more than two letters in the teambuilder search; this PR introduces debounced calls to updateTeamList() instead of direct calls.

Tested locally on Chrome 127.0.6533.72

### CODE REVIEW

**Comment by Zarel** [comment]
Ten years ago this was instantaneous enough, and computers have only gotten better since then. Probably this should at least be a setting?

**Comment by PartMan7** [comment]
Takes about 5-10s per character input on my device (which has fairly decent specs); this isn't an issue for people who don't store teams, but for people with a lot (I'm around 9K or so), it is... problematic, to say the least. I'm not sure if there's any benefit to _not_ having it debounced (except the removal of the 300ms typing delay), but it offers pretty noticeable benefits for users that do.

File: data/pokemon-showdown
Comment (PartMan7) [inline data/pokemon-showdown:1]: Wait just saw this; will remove

**Comment by Zarel** [comment]
Why does this only get laggy at a high team count? Is the team list still hidden somewhere in the DOM?

**Comment by Zarel** [comment]
Probably _that's_ the bug to fix.

**Comment by Zarel** [comment]
Thanks! This is a good stopgap until we get a real fix in.

**Pull request merged**